pkg/api/v1/control-plane/cache/prioritylist.go

```diff
@@ -0,0 +1,165 @@
+package cache
+
+import (
+	"fmt"
+	"sort"
+)
+
+// The default Priority is the last index. The last index is not required in the prioirty list, thus any elements added
+// to the struct will be added to the default priority (last).
+const defaultPrioirty = -1
+
+type PriorityValue interface {
+	GetPriority() string
+}
+
+// PrioritySortedStruct sorts elements by prioirity. Priority lists are unordered.
+// Needs a priority list and a get priority function.
+// If a priority does not exist for an added element, it is added to the lowest priority.
+type PrioritySortedStruct struct {
+	// prioritySets is the set lists for priorities, where P is the type used for priority.
+	prioritySets map[int][]string
+	// priorityMap maps the value to the priority.
+	priorityMap map[string]int
+	// elements are the map of elements structured by their priority.
+	elements []map[int]ResponseWatch
+	// numberOfPriorities is the number of priorities in the prioritySets
+	numberOfPriorities int
```

Comment:
Why do we even care about this? isnt this  just the len of prioritymap?

Reply:
I think it makes it easy to read

Reply:
We can take this out, since so many people have issues with variable here. It can be retrieved with `len(p.prioritySet)`